atlas-2024-odfr/mkjson.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+
+output_directory = 'test'
+import os
+try:
+    os.mkdir(output_directory)
+except:
+    pass
+
+'''
+Notes on the plan:
+
+-) One record for the 2015 data
+	- Structure in the record to represent the individual runs
+-) One record for the 2016 data
+	- Structure in the record to represent the individual runs
+-) One record for each of the "blocks" of MC that we had in the note (https://cds.cern.ch/record/2875752)
+	- 9 tables in total for the time being
+	- Structure in the record to represent the individual MC datasets
+
+You can see an example of this kind of structure in the "File Indexes" section of this record:
+https://opendata.cern.ch/record/24464
+
+I'd then write a draft _very short_ description for each of the 11 pages, linking to the main documentation
+and explaining in a few sentences what that particular record represents. We would give each record a DOI,
+and we could create a `meta record' containing links to all 11 pages that summarize the full release of Open Data.
+'''
+
+import json
+
+# Get datasets
+dataset_files = {
+  'pp_2015_data_p6026_tids.txt':'Run 2 2015 proton-proton collision data',
+  'pp_2016_data_p6026_tids.txt':'Run 2 2016 proton-proton collision data',
+  'mc_boson_nominal.txt':'MC simulation electroweak boson nominal samples',
+  'mc_exotics_nominal.txt':'MC simulation exotic signal samples',
+  'mc_higgs_nominal.txt':'MC simulation Higgs nominal samples',
+  'mc_higgs_systematics.txt':'MC simulation Higgs systematic variation samples',
+  'mc_jet_nominal.txt':'MC simulation QCD jet nominal samples',
+  'mc_jet_systematics.txt':'MC simulation QCD jet systematic variation samples',
+  'mc_susy_nominal.txt':'MC simulation SUSY signal samples',
+  'mc_top_nominal.txt':'MC simulation top nominal samples',
+  'mc_top_systematics.txt':'MC simulation top systematic variation samples'
+    }
+
+# Populate fields
+
+# This is applicable for the pp data only!
+evergreen_data = {
+    # Accelerator - just CERN LHC
+    "accelerator": "CERN-LHC",
+    # ATLAS Collaboration; recid only if we need a specific author list
+    "collaboration": {
+      "name": "ATLAS collaboration",
+     },
+    # Basic collision data - this applies only to the pp data
+    "collision_information": {
+      "energy": "13TeV",
+      "type": "pp"
+    },
+    # Published this year!
+    "date_published": "2024",
+    # ATLAS experiment
+    "experiment": [
+      "ATLAS"
+    ],
+    # Thanks to the Open Data Portal
+    "publisher": "CERN Open Data Portal",
+    # Note: beginning of the reprocessing
+    "date_reprocessed": "2020",
+    "distribution": {
+      "formats": [
+        "DAOD_PHYSLITE"
+      ],
+    },
+    # Dataset type information for Open Data Portal
+    "type": {
+      "primary": "Dataset",
+      "secondary": [
+        "Collision"
+      ]
```

Comment:
Please add record ID field which is mandatory for the produced JSON files to be loaded as open data repository records.  The field should look like:

```json
    "recid": "80000",
```

You can use record ID values starting from 80000.

BTW the record IDs are persistent, so it may be nice to keep some correspondence such as:

```python
record_ids = {
    "pp_2015_data_p6026_tids.txt": 80000,
    "pp_2016_data_p6026_tids.txt": 80001,
    "mc_boson_nominal.txt": 80010,
    "mc_exotics_nominal.txt": 80011,
    "mc_higgs_nominal.txt": 80012,
    "mc_higgs_systematics.txt": 80013,
    "mc_jet_nominal.txt": 80014,
    "mc_jet_systematics.txt": 80015,
    "mc_susy_nominal.txt": 80016,
    "mc_top_nominal.txt": 80017,
    "mc_top_systematics.txt": 80018,
}
```

Assuming the keys such as "mc_exotics_nominal.txt" are your unique identifiers for a dataset, similarly to the `dataset_files` variable in `mkjson.py`.

(Note that we shall have the same need for the DOI persistent identifier that this script will eventually create too.)

Reply:
Done, and essentially in the manner suggested (except I'm keeping a list with all the properties I want... ok, a dictionary would be nicer, but this is clear enough I hope.